src/PKSim.Infrastructure/InfrastructureRegister.cs

```diff
@@ -124,25 +129,50 @@ public static void RegisterSerializationDependencies()
          container.Register(typeof(IXmlReader<>), typeof(XmlReader<>));
          container.Register(typeof(IXmlWriter<>), typeof(XmlWriter<>));
 
-         //load repository to trigger initialization
+         //Core serialization to serialize to pkml compatible with core
+         container.AddRegister(x => x.FromType<CoreSerializerRegister>());
+      }
+
+      public static void LoadDefaultEntities(IContainer container)
+      {
+         //load unit serializer
+         container.Resolve<IUnitSystemXmlSerializerRepository>().PerformMapping();
+         //Then trigger dimension load to load the dimensions
+         loadDimensionRepository(container);
+
+         //Load pkml serializers
+         var ospSuiteXmlSerializerRepository = container.Resolve<IOSPSuiteXmlSerializerRepository>();
+         ospSuiteXmlSerializerRepository.AddPresentationSerializers();
+         ospSuiteXmlSerializerRepository.PerformMapping();
+         //then load pk parameters
+         loadPKParameters(container);
+
+         //last load all serializer
          container.Resolve<IPKSimXmlSerializerRepository>().PerformMapping();
+      }
 
+      private static void loadDimensionRepository(IContainer container)
+      {
+         //Loading the dimension will trigger the load
+         container.Resolve<IDimensionRepository>().All();
+      }
+
+      private static void loadPKParameters(IContainer container)
+      {
          //register pk analyses values
          var pkParameterRepository = container.Resolve<IPKParameterRepository>();
          var pKParameterLoader = container.Resolve<IPKParameterRepositoryLoader>();
          var pkSimConfiguration = container.Resolve<IPKSimConfiguration>();
          pKParameterLoader.Load(pkParameterRepository, pkSimConfiguration.PKParametersFilePath);
       }
 
-      public static void RegisterWorkspace()
+      public static void RegisterWorkspace(IContainer container)
       {
-         var container = IoC.Container;
          container.Register<IWorkspace, IWithWorkspaceLayout, ICoreWorkspace, OSPSuite.Core.IWorkspace, Workspace>(LifeStyle.Singleton);
       }
 
-      private void registerORMDependencies()
+      private void registerORMDependencies(IContainer container)
       {
-         var container = IoC.Container;
```

Comment:
I cannot use IoC...at all. Using ospsuite-R already populates the global IoC and for some reason, PKSim is in the same context. Nothing was working when loading PKSim and OSPSuite because of the shared container (being even two dffieretn components as PKSim is using castle and OSPSuite.R is using autofac)

So now I am not using IoC at all in PKSim.R which allows me to create a new instance of the container